subprojects/core/src/main/java/org/gradle/api/internal/initialization/transform/utils/InstrumentationClasspathMerger.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.initialization.transform.utils;
+
+import com.google.common.collect.Ordering;
+import org.gradle.api.artifacts.ArtifactCollection;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.artifacts.result.ResolvedArtifactResult;
+import org.gradle.internal.component.local.model.TransformedComponentFileArtifactIdentifier;
+
+import java.io.File;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class InstrumentationClasspathMerger {
```

Comment:
This class is just some code extracted from DefaultScriptClassPathResolver.

Reply:
The only change is that now we use `List<File>` to avoid deduplication, right?

Reply:
Yes, and toString methods on classes, but that I used just for debugging and left it here